engine/src/ui/canvas/plugin.cpp

```diff
@@ -116,4 +134,58 @@ void cubos::engine::uiCanvasPlugin(Cubos& cubos)
                 window->renderDevice().clearDepth(1);
             }
         });
+
+    cubos.system("draw ui")
+        .tagged(uiEndTag)
+        .with<UIElement>()
+        .related<ChildOf>()
+        .with<UICanvas>()
+        .other()
+        .with<UIElement>()
+        .related<ChildOf>()
+        .with<UIElement>()
+        .call([](Query<Entity, UIElement&, const UICanvas&> drawsToQuery,
+                 Query<Entity, UIElement&, UIElement&> childrenQuery,
+                 Query<Entity, UICanvas&, RenderTarget&> canvasQuery, const core::io::Window& window) {
+            cubos::core::gl::DepthStencilStateDesc dssd;
+            dssd.depth.enabled = true;
+            dssd.depth.writeEnabled = true;
+            dssd.depth.near = -1.0F;
+            dssd.depth.compare = Compare::LEqual;
+            window->renderDevice().setDepthStencilState(window->renderDevice().createDepthStencilState(dssd));
```

Comment:
Create this once in a startup system and just bind it here (keep it on a `State` resource ig).